crates/oxc_linter/src/rules/eslint/no_control_regex.rs

```diff
@@ -0,0 +1,351 @@
+            for m in control_patterns(pattern) {
+                let ctl = m.as_str();
+
+                // check for an even number of backslashes, since these will
+                // prevent the pattern from being a control sequence
+                if ctl.starts_with('\\') && m.start() > 0 {
+                    let pattern_chars: Vec<char> = pattern.chars().collect(); // ew
+
+                    let mut first_backslash = m.start();
+                    while first_backslash > 0 && pattern_chars[first_backslash] == '\\' {
+                        first_backslash -= 1;
+                    }
+
+                    let mut num_slashes = m.start() - first_backslash;
+                    if first_backslash == 0 && pattern_chars[first_backslash] == '\\' {
+                        num_slashes += 1;
+                    }
+                    // even # of slashes
+                    if num_slashes % 2 == 0 {
+                        continue;
+                    }
+                }
+
+                if ctl.starts_with(r"\x") || ctl.starts_with(r"\u") {
+                    let mut numeric_part = &ctl[2..];
+
+                    // extract numeric part from \u{00}
+                    if numeric_part.starts_with('{') {
+                        let has_unicode_flag = match flags {
+                            Some(flags) if flags.contains(RegExpFlags::U) => true,
+                            _ => {
+                                continue;
+                            }
+                        };
+
+                        // 1. Unicode control pattern is missing a curly brace
+                        //    and is therefore invalid. (note: we may want to
+                        //    report this?)
+                        // 2. Unicode flag is missing, which is needed for
+                        //    interpreting \u{`nn`} as a unicode character
+                        if !has_unicode_flag || !numeric_part.ends_with('}') {
+                            continue;
+                        }
+
+                        numeric_part = &numeric_part[1..numeric_part.len() - 1];
+                    }
+
+                    match u32::from_str_radix(numeric_part, 16) {
+                        Err(_) => continue,
+                        Ok(as_num) if as_num > 0x1f => continue,
+                        Ok(_) => { /* noop */ }
+                    }
+                }
+
+                violations.push(ctl);
+            }
+
+            if !violations.is_empty() {
+                let violations = violations.join(", ");
+                context.diagnostic(NoControlRegexDiagnostic(violations.into(), span));
+            }
+        }
+    }
+}
+
+fn extract_flags<'a>(args: &'a oxc_allocator::Vec<'a, Argument<'a>>) -> Option<RegExpFlags> {
+    if args.len() > 1 &&
+    let Argument::Expression(Expression::StringLiteral(flag_arg)) = &args[1] {
+        // TODO: how should we handle invalid flags?
```

Comment:
These are syntax errors, see https://github.com/Boshen/oxc/blob/ed9f67d263390c5607342c249d2a1f898502b6c9/crates/oxc_parser/src/lexer/mod.rs#L1072-L1077

Reply:
I agree. Do you think reporting them is outside the scope of this rule?

Reply:
I'm going to resolve this if there aren't any action items.